tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingFlakyTests.cs

```diff
@@ -0,0 +1,149 @@
+{
+    public abstract class InfiniteTracingFlakyTestsBase<TFixture> : NewRelicIntegrationTest<TFixture>
+        where TFixture : ConsoleDynamicMethodFixture
+    {
+        private readonly TFixture _fixture;
+
+        public InfiniteTracingFlakyTestsBase(TFixture fixture, ITestOutputHelper output) : base(fixture)
+        {
+            _fixture = fixture;
+            _fixture.TestLogger = output;
+            _fixture.SetTimeout(System.TimeSpan.FromMinutes(5));
+
+            // Ensure the trace observer will throw an error on every request
+            _fixture.RemoteApplication.SetAdditionalEnvironmentVariable("NEW_RELIC_INFINITE_TRACING_SPAN_EVENTS_TEST_FLAKY", "100");
+
+            // set the code to be returned when the trace observer throws an error
+            // must be a value from 0 to 16 as per https://github.com/grpc/grpc/blob/master/doc/statuscodes.md
+            // use the StatusCode enum from gRPC 
+            _fixture.RemoteApplication.SetAdditionalEnvironmentVariable("NEW_RELIC_INFINITE_TRACING_SPAN_EVENTS_TEST_FLAKY_CODE",
+                ((int)Grpc.Core.StatusCode.Internal).ToString());
+
+            _fixture.AddCommand("InfiniteTracingTester StartAgent");
+
+            _fixture.AddCommand("RootCommands DelaySeconds 15"); // give the agent time to warm up
+
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+
+            _fixture.AddActions
+            (
+                setupConfiguration: () =>
+                {
+                    var configModifier = new NewRelicConfigModifier(fixture.DestinationNewRelicConfigFilePath);
+
+                    configModifier
+                        .ForceTransactionTraces()
+                        .EnableDistributedTrace()
+                        .EnableInfiniteTracing(_fixture.TestConfiguration.TraceObserverUrl, _fixture.TestConfiguration.TraceObserverPort)
+                        .SetLogLevel("finest");
+                }
+                ,
+                exerciseApplication: () =>
+                {
+                    // wait up to 75 seconds for the harvest cycle to complete and emit the supportability metrics we're expecting
+                    var waitUntil = DateTime.Now.AddSeconds(75);
+                    while (DateTime.Now <= waitUntil
+                           && !_fixture.AgentLog.GetMetrics().Any(metric => metric.MetricSpec.Name == "Supportability/InfiniteTracing/Span/gRPC/INTERNAL")
+                           && !_fixture.AgentLog.GetMetrics().Any(metric => metric.MetricSpec.Name == "Supportability/InfiniteTracing/Span/Response/Error"))
```

Comment:
```suggestion
                           && !(_fixture.AgentLog.GetMetrics().Any(metric => metric.MetricSpec.Name == "Supportability/InfiniteTracing/Span/gRPC/INTERNAL")
                           && _fixture.AgentLog.GetMetrics().Any(metric => metric.MetricSpec.Name == "Supportability/InfiniteTracing/Span/Response/Error")))
```

The existing code would return when one or the other metric was detected. Though, it's not like we'd get a second metric harvest in 75 seconds just waiting...